src/coreclr/jit/fgopt.cpp

```diff
@@ -132,19 +132,17 @@ bool Compiler::fgReachable(BasicBlock* b1, BasicBlock* b2)
 
     if (b1->bbNum > fgDomBBcount)
     {
-        noway_assert(b1->KindIs(BBJ_NONE, BBJ_ALWAYS, BBJ_COND));
+        noway_assert(b1->KindIs(BBJ_ALWAYS, BBJ_COND));
 
-        if (b1->KindIs(BBJ_NONE, BBJ_COND) && fgReachable(b1->Next(), b2))
+        if (b1->KindIs(BBJ_COND))
         {
-            return true;
+            return fgReachable(b1->Next(), b2) || fgReachable(b1->GetJumpDest(), b2);
         }
-
-        if (b1->KindIs(BBJ_ALWAYS, BBJ_COND) && fgReachable(b1->GetJumpDest(), b2))
+        else
         {
-            return true;
+            noway_assert(b1->KindIs(BBJ_ALWAYS));
```

Comment:
This assert seems unnecessary. At least, make it a normal `assert` (not a `noway_assert`)